io.openems.edge.ess.adstec.storaxe/src/io/openems/edge/ess/adstec/storaxe/AdstecStoraxeEssImpl.java

```diff
@@ -0,0 +1,142 @@
+				config.modbus_id());
+		this._setCapacity(config.capacity());
+	}
+
+	@Deactivate
+	protected void deactivate() {
+		super.deactivate();
+	}
+
+	@Override
+	protected ModbusProtocol defineModbusProtocol() throws OpenemsException {
+		final int offset = -1; // The Modbus library seems to use 0 offsets.
+		return new ModbusProtocol(this, //
+				new FC4ReadInputRegistersTask(1 + offset, Priority.LOW, //
+						m(SymmetricEss.ChannelId.GRID_MODE, new UnsignedWordElement(1 + offset), GRID_MODE_CONVERTER),
+						m(SymmetricEss.ChannelId.ACTIVE_POWER, new SignedWordElement(2 + offset),
+								ElementToChannelConverter.SCALE_FACTOR_2),
+						m(SymmetricEss.ChannelId.REACTIVE_POWER, new SignedWordElement(3 + offset),
+								ElementToChannelConverter.SCALE_FACTOR_2)),
+				new FC4ReadInputRegistersTask(125 + offset, Priority.LOW, //
+						m(SymmetricEss.ChannelId.MAX_APPARENT_POWER, new UnsignedWordElement(125 + offset),
+								ElementToChannelConverter.SCALE_FACTOR_2),
+						m(SymmetricEss.ChannelId.SOC, new UnsignedWordElement(126 + offset),
+								ElementToChannelConverter.DIRECT_1_TO_1)),
+				new FC4ReadInputRegistersTask(134 + offset, Priority.LOW, //
+						m(SymmetricEss.ChannelId.ACTIVE_DISCHARGE_ENERGY, new UnsignedDoublewordElement(134 + offset),
+								ElementToChannelConverter.SCALE_FACTOR_3),
+						m(SymmetricEss.ChannelId.ACTIVE_CHARGE_ENERGY, new UnsignedDoublewordElement(136 + offset),
+								ElementToChannelConverter.SCALE_FACTOR_3),
+						m(SymmetricEss.ChannelId.MIN_CELL_VOLTAGE, new UnsignedWordElement(138 + offset),
+								ElementToChannelConverter.DIRECT_1_TO_1),
+						m(SymmetricEss.ChannelId.MAX_CELL_VOLTAGE, new UnsignedWordElement(139 + offset),
+								ElementToChannelConverter.DIRECT_1_TO_1)) //
+		);
+	}
+
+	@Override
+	public String debugLog() {
+		return "SoC:" + this.getSoc().asString() //
+				+ "|L:" + this.getActivePower().asString(); //
+	}
+
+	@Override
+	public ModbusSlaveTable getModbusSlaveTable(AccessMode accessMode) {
+		return new ModbusSlaveTable(//
+				OpenemsComponent.getModbusSlaveNatureTable(accessMode), //
+				SymmetricEss.getModbusSlaveNatureTable(accessMode), //
+				AsymmetricEss.getModbusSlaveNatureTable(accessMode), //
+				ModbusSlaveNatureTable.of(AdstecStoraxeEss.class, accessMode, 300) //
+						.build());
+	}
+
+	@Override
+	public Value<Integer> getReactivePower() {
+		// We need to override because the ess returns neg for capacitative, pos for
+		// inductive, but the channel expects
+		// pos for from-battery, neg for to-battery.
+		return new Value<Integer>(this.getReactivePowerChannel(),
+				Math.abs(this.getReactivePowerChannel().value().get()) * Integer.signum(this.getActivePower().get()));
+	}
+
+	private static final ElementToChannelConverter GRID_MODE_CONVERTER = new ElementToChannelConverter(value -> {
+		switch ((Integer) value) {
+		case 1:
+			return 2;
+		case 2:
+			return 1;
+		default:
+			return 0;
+		}
+	}, value -> {
+		switch ((Integer) value) {
```

Comment:
Reverse is not strictly required here, because GridMode can never be set.

Reply:
Yeah, great, I thought we needed to create both. I just saw the implementation for one-way. Thanks!